rioxarray/rioxarray.py

```diff
@@ -1078,3 +1079,78 @@ def transform_bounds(self, dst_crs, densify_pts=21, recalc=False):
         return rasterio.warp.transform_bounds(
             self.crs, dst_crs, *self.bounds(recalc=recalc), densify_pts=densify_pts
         )
+
+    def write_gcps(self, gcps, grid_mapping_name=None, inplace=False):
+        """
+        Write the GroundControlPoints to the dataset.
+
+        https://rasterio.readthedocs.io/en/latest/topics/georeferencing.html#ground-control-points
+
+        Parameters
+        ----------
+        gcp: (list of rasterio.GroundControlPoints, corresponding crs)
+            The Ground Control Points to integrate to the dataset.
+        grid_mapping_name: str, optional
+            Name of the grid_mapping coordinate to store the GCPs information in.
+            Default is the grid_mapping name of the dataset.
+        inplace: bool, optional
+            If True, it will write to the existing dataset. Default is False.
+
+        Returns
+        -------
+        :obj:`xarray.Dataset` | :obj:`xarray.DataArray`:
+            Modified dataset with Ground Control Points written.
+        """
+        grid_mapping_name = (
+            self.grid_mapping if grid_mapping_name is None else grid_mapping_name
+        )
+        data_obj = self._get_obj(inplace=True)
+
+        gcps, gcp_crs = gcps
+        data_obj = data_obj.rio.write_crs(gcp_crs, inplace=inplace)
```

Comment:
Probably a good idea to pass in `grid_mapping_name` here.

Reply:
Fixed